repose-aggregator/components/filters/openstack-identity-v3-filter/src/main/scala/org/openrepose/filters/openstackidentityv3/OpenStackIdentityV3Handler.scala

```diff
@@ -87,6 +86,19 @@ class OpenStackIdentityV3Handler(identityConfig: OpenstackIdentityV3Config, iden
       response.addHeader(HttpHeaders.RETRY_AFTER, retry)
     }
 
+    implicit val catalogWrites = new Writes[ServiceForAuthenticationResponse] {
+      def writes(catalog: ServiceForAuthenticationResponse) = JsObject(Seq(
+        "endpoints" -> JsArray(catalog.endpoints.map(endpoint => JsObject(
+          Seq("id" -> JsString(endpoint.id)) ++
+            endpoint.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()) ++
+            endpoint.interface.map(interface => Seq("interface" -> JsString(interface))).getOrElse(Seq()) ++
+            endpoint.region.map(region => Seq("region" -> JsString(region))).getOrElse(Seq()) ++
+            Seq("url" -> JsString(endpoint.url)) ++
+            endpoint.service_id.map(serviceId => Seq("service_id" -> JsString(serviceId))).getOrElse(Seq()))))) ++
+        catalog.id.map(id => Seq("id" -> JsString(id))).getOrElse(Seq()) ++
+        catalog.name.map(name => Seq("name" -> JsString(name))).getOrElse(Seq()))
+    }
```

Comment:
It is also odd that line 92 and line 93 are indented differently. They are sibling fields in a JSON object, right?